### PR TITLE
Change upload from bytes to Stream

### PR DESCRIPTION
**Pull request by sherlockvn**

### CODE REVIEW

**Comment by nqhhdev** [comment]
- IMO: Pls create small demo for `UploadFileAPI`

File: lib/data/network/upload_file/upload_file_api.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    final responseBody = await response.stream.toBytes();
+    if (response.statusCode != 200) unexpectedResponse(response, responseBody);
+    final responseString = utf8.decode(responseBody);
+    final json = jsonDecode(responseString);
+    return Uri.parse(json['content_uri']);
+  }
+}
+
+extension UploadFileAsMultipart on Client {
+
```
Comment (imGok) [inline lib/data/network/upload_file/upload_file_api.dart:42]:
```suggestion

```

File: lib/data/network/upload_file/upload_file_api.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    final maxMediaSize = mediaConfig.mUploadSize;
+    final fileLength = await file.length();
+    if (maxMediaSize != null && maxMediaSize < fileLength) {
+      throw FileTooBigMatrixException(fileLength, maxMediaSize);
+    }
+
+    contentType ??= lookupMimeType(filename ?? '');
+    final mxc = await uploadContentByMultipartType(
+      file: file,
+      filename: filename,
```
Comment (imGok) [inline lib/data/network/upload_file/upload_file_api.dart:55]: remove trailing whitespaces
```suggestion
      filename: filename,
```

File: lib/data/network/upload_file/upload_file_api.dart [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    if (response.statusCode != 200) unexpectedResponse(response, responseBody);
+    final responseString = utf8.decode(responseBody);
+    final json = jsonDecode(responseString);
+    return Uri.parse(json['content_uri']);
+  }
+}
+
+extension UploadFileAsMultipart on Client {
+  
+  Future<Uri> uploadContentAsMultipart(File file,
```
Comment (imGok) [inline lib/data/network/upload_file/upload_file_api.dart:43]: where is it used ?

**Pull request closed without merging**